bazarr/subtitles/refiners/anidb.py

```diff
@@ -129,7 +129,15 @@ def refine_anidb_ids(video):
 
     season = video.season if video.season else 0
 
-    anidb_series_id, anidb_episode_id = anidb_client.get_series_episodes_ids(video.series_tvdb_id, season, video.episode)
+    series_episode_ids = anidb_client.get_series_episodes_ids(video.series_tvdb_id, season, video.episode)
+
+    # Account for region-cache having stored outdated code of get_series_episodes_ids()
+    if len(series_episode_ids) == 3:
+        anidb_series_id, anidb_episode_id, anidb_episode_no = series_episode_ids
+    else:
+        logging.warn("Could not determine actual episode number of this video due to region-cache.")
+        anidb_series_id, anidb_episode_id = series_episode_ids
+        anidb_episode_no = None
```

Comment:
I'm trying to improve this refiner because of the daily limit of 200, we will need to cache for more time, and handle api banned as well, i think relying in the length is not ideal, we should raise an exception inside the refiner and catch different things here, this is what im planning to do in another PR:

```
 try:
        anidb_series_id, anidb_episode_id = anidb_client.get_series_episodes_ids(
            video.series_tvdb_id,
            season, video.episode,
        )
    except TooManyRequests:
        logging.error(f'API daily limit reached while refining {video.series}')

        region.set(cache_key, "dwqdwq")

        return video
```

Reply:
Perhaps if the episode is not found, raise a different exception and handle it instead of rely in the length?

Reply:
Also on the line 80 you should return None,None,None

Reply:
You can also check if anidb_episode_no is None in that case.

Reply:
The prime purpose of this code is to handle cases where the Jimaku provider would be used on series that have used the AnimeTosho provider earlier, therefore having run the AniDB refiner already.  
As `get_series_episodes_ids()` has been cached for such series, a tuple of length 2 is returned, resulting in errors when `refine_anidb_ids()` is doing its thing as it now expects a tuple with 3 items.
That's why I'm checking the length.  

> You can also check if anidb_episode_no is None in that case.

Already doing so in the provider, unless you're suggesting to implement such a check in the refiner?

> Also on the line 81 you should return None,None,None.  

Somehow missed that line... Always returning an equal amount of arguments obviously makes more sense.